### DISCUSSION TITLE
wxMaxima Becomes Confused on Filenames

### DISCUSSION THREAD

**Discussion by Donatien-Alphonse** [wxMaxima-developers/wxmaxima#1610] [Q&A]
Greetings, all,

I just noticed the following anomaly in wxMaxima version 20.12.2 (Maxima 5.44.0), and don't remember encountering it in previous versions. I'm using Linux Mint 19.3.

Problem: Suppose you've got a folder named "My Examples", and within it is a file named "Problem 1.wxmx".

Then in the Nemo desktop file manager window, you're looking at the icon for that file. You double-click on it, expecting wxMaxima to open and bring in the file /My Examples/Problem 1.wxmx, in the usual way we're all accustomed to with GUI's.

Instead, four instances of wxMaxima are started, each one attempting to open the following erroneous files, respectively:

My.wxmx
Examples.wxmx,
Problem.wxmx
1.wxmx

Of course, there are no such files, so each instance of wxMaxima halts with an error message that the file can't be found. One has to tediously click OK on the four error messages one-by-one to close them out, and then instead start wxMaxima first and open the file from its file menu. This is not very user-friendly.

So, is this a bug, or is there a secret setting I'm not familiar with to get wxMaxima to understand folder names and filenames properly?

Many thanks,

Donatien-Alphonse

It appears that wxMaxima

**Comment by daute** [comment]
I assume, thats a bug in the file manager. It should call
`wxmaxima "My Examples/Problem 1.wxmx"`
but does **not** quote the argument. Most likely that occurs with other file types too?

**Comment by Donatien-Alphonse** [comment]
I've been waiting this bug out, hoping  version  22.05.0 of wxMaxima would have fixed it. But no such luck. The program still becomes confused if the filename or the path of a workbook file contains spaces. I'm using Linux Mint 20.3 now.

This is an extremely annoying bug. No other programs in Linux Mint exhibit this behavior. Is there any chance this will be addressed in the future? I would hate to have to go back to an early version of wxMaxima just to avoid it.

Thanks for listening!

Donatien-Alphonse

**Comment by gunterkoenigsmann** [comment]
I use paths with spaces all the time. What seems to be the problem?

Kind regards,

   Gunter.

Am 31. August 2022 03:43:44 MESZ schrieb Donatien-Alphonse ***@***.***>:

**Comment by Donatien-Alphonse** [comment]
Hello Gunter,

Thanks for your response. Here's an example. Suppose I want to double-click to open this worksheet in wxMaxima:

/home/studs/Documents/Mathematics/wxMaxima/Mathematica Examples in wxMaxima/Chapter 1.wxmx

In the folder it looks like this:

![Error1](https://user-images.githubusercontent.com/93812232/187607820-76f2c164-a3c3-4127-90db-8ba245845427.jpg)

But then wxMaxima stumbles upon the spaces and attempts to open multiple files based upon the space-separated words in the path or filename:

![Error2](https://user-images.githubusercontent.com/93812232/187608149-122e4087-a22a-4246-a771-73a414d3539d.jpg)

For emphasis, this doesn't happen with version 19 of wxMaxima, only with newer versions, and in either Linux Mint 19.3 or Linux Mint 20.3.

Since it doesn't happen for you, any idea what I'm doing wrong? Thanks so much for your interest.

Best wishes,

Donatien-Alphonse

**Comment by gunterkoenigsmann** [comment]
Tested filenames with spaces in them on my computer - and on my computer they seem to work, still.

Normally double-clicking a file that wxMaxima can open does nothing but starting the program wxMaxima with the filename you double-clicked on as argument. What the output from wxMaxima from your screenshot seems to indicate that whatever program you clicked on doesn't pass the filename as an argument, but rather does interpret the spaces as filename separators. Does double-clicking on any other file (text file, html file or similar) with spaces in its name work on your computer?

**Comment by Donatien-Alphonse** [comment]
I have no problems with any other programs (some two dozen I use on a regular basis). Double-clicking on .txt, .html, files etc. opens them as expected in the associated program.

It occurs to me this may have something to do with the fact the wxMaxima I'm using is an AppImage.

More specifically: previous version of wxMaxima (version 19.xx) is not an AppImage and handles file names perfectly. But version 22,xx is AppImage and stumbles on spaces.

Could that be the problem?

**Comment by Donatien-Alphonse** [comment]
As an experiment, I downloaded another program which is in AppImage format (Kdenlive video editor). It too has difficulties if attempting to run by double-clicking an associated file.

I suspect then that the problem is not so much with wxMaxima, but rather with how its AppImage behaves in Linux Mint.

Any thoughts from you all?

Donatien-Alphonse
